presto-docs/src/main/sphinx/connector/bigquery.rst

```diff
@@ -0,0 +1,156 @@
+
+Requirements
+------------
+
+Enable the BigQuery Storage API
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Follow `these instructions <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.
+
+Authentication
+^^^^^^^^^^^^^^
+**On GCE/Dataproc** the authentication is taken from the machine's role.
+
+**Outside GCE/Dataproc** you have 3 options:
+
+* Use a service account JSON key and ``GOOGLE_APPLICATION_CREDENTIALS`` as described `here <https://cloud.google.com/docs/authentication/getting-started>`_.
+* You can set the ``credentials`` key in the catalog properties file. It should be set to the JSON contents, converted using base64: ``credentials=<SERVICE_ACCOUNT_JSON_IN_BASE64>``
+* You can set the ``credentials-file`` key in the catalog properties file. It should point to the location of the JSON file ``credentials-file=</path/to/key/file>``
+
+Configuration
+-------------
+
+To configure the BigQuery connector, create a catalog properties file
+in ``etc/catalog`` named, for example, ``bigquery.properties``, to
+mount the BigQuery connector as the ``bigquery`` catalog.
+Create the file with the following contents, replacing the
+connection properties as appropriate for your setup:
+
+.. code-block:: none
+
+    connector.name=bigquery
+    project-id=<your Google Cloud Platform project id>
+
+
+
+
+Multiple GCP Projects
+^^^^^^^^^^^^^^^^^^^^^
+
+The BigQuery connector can only access a single GCP project.Thus, if you have
+data in multiple GCP projects, You need to create several catalogs, each pointing
+to a different GCP project. For example, if you have two GCP projects, one for
+the sales and one for analytics, you can create two properties files in ``etc/catalog``
+named ``sales.properties`` and ``analytics.properties``, both having
+``connector.name=bigquery`` but with different ``project-id``. This will create the two
+catalogs, ``sales`` and ``analytics`` respectively.
+
+Configuring Partitioning
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default the connector creates one partition per 400MB in the table being read (before filtering). This should roughly correspond to the maximum number of readers supported by
+the BigQuery Storage API. This can be configured explicitly with the ``parallelism`` property. BigQuery may limit the number of partitions based on server constraints.
+
+Reading From Views
+^^^^^^^^^^^^^^^^^^
+The connector has a preliminary support for reading from `BigQuery views <https://cloud.google.com/bigquery/docs/views-intro>`_. Please note there are a few caveats:
+
+* BigQuery views are not materialized by default, which means that the connector needs to materialize them before it can read them. This process affects the read performance.
+* The materialization process can also incur additional costs to your BigQuery bill.
+* By default, the materialized views are created in the same project and dataset. Those can be configured by the optional ``view-materialization-project``
```

Comment:
Prefix all the config names with `bigquery.` to match the actual property name (so users can copy/paste)